src/Inputs/Slider.tsx

```diff
@@ -1,80 +1,351 @@
+    popupleft=-17,
+    popuptop=-51,
+    popupwidth="auto",
+    popupheight=30,
+    ...props
+}): React.ReactElement =>{  
+
+    const barRef = useRef() as RefObject<HTMLDivElement>;
+    
+    const translateToPixels = (theValue:number|undefined, scale:number|undefined, maximum:number|undefined, minimum:number) : string =>{
+        const maxmindifference = (maximum as number - minimum as number) as number;
+        const pixelTranslator = scale as number / maxmindifference;
+
+        return ((theValue as number - minimum as number) * pixelTranslator) + "px";
+    }
+
+    const translateToValue=(theValue:number, scale:number, maximum:number|undefined, minimum:number, step:number)=>{
+        return Math.round((((theValue * (maximum as number - minimum)) / scale) + minimum) / step) * step;
+    }
+
+    const calculatePosition = (theValue:number, scale:number, maximum:number|undefined, minimum:number, step:number) =>{
+        return translateToPixels(translateToValue(theValue,scale,maximum,minimum,step),scale,maximum as number|undefined,minimum);
+    }
+    useLayoutEffect(():void => {
+        if (null !== barRef.current){
+            const bar = barRef.current as HTMLElement;
+
+            const selectedBar = bar.firstChild as HTMLElement;
+            const finishThumb = bar.lastChild as HTMLElement;
+            const startThumb = finishThumb.previousElementSibling as HTMLElement;
+            const childrenbar = bar.nextElementSibling as HTMLElement;
+
+            const clientwidth = bar.clientWidth;
+
+            const maxinpixels = translateToPixels(max,clientwidth,max,min as number);
+                    
+
+
+            finishThumb.style.left = translateToPixels(value,clientwidth,max,min as number);
+            startThumb.style.left = translateToPixels(valuestart,clientwidth,max,min as number);
+            selectedBar.style.left = startThumb.style.left;
+            selectedBar.style.width = twoinputs ? (parseInt(finishThumb.style.left) - parseInt(startThumb.style.left)) +"px" :(parseInt(finishThumb.style.left)) + "px";
+
+            bar.style.width = maxinpixels;
+            childrenbar.style.width = bar.style.width;
+
+
+            const theChildren = childrenbar.children as any ;
+
+            theChildren.forEach((child:HTMLElement)=>{
+                console.log(typeof(child));
+                child.style.left=translateToPixels(parseInt(child.style.left),clientwidth,max,min as number);
+            });
+
+            finishThumb.style.zIndex = "1";
+            startThumb.style.zIndex = "1";
+        }
+    },[]);
+
+    const handleMouseDown=(e:MouseEvent)=>{
+        if (!disabled) {
+        
+          let thumb = e.target as HTMLElement;
+          let bar = thumb.parentElement as HTMLElement;
+          let selectedbar = bar.firstChild as HTMLElement;
+          let shiftX = e.clientX - thumb.getBoundingClientRect().left;
+
+          document.addEventListener('mousemove', onMouseMove as any);     

```

Comment:
Event listeners should be added and removed on component mount/unmount, not on an event